pandas/core/groupby/generic.py

```diff
@@ -1232,20 +1226,19 @@ def first_not_none(values):
                     key_index = None
 
             # make Nones an empty object
-            v = first_not_none(values)
```

Comment:
it looks like there are two places where first_not_none is called, and `values` can change in between them

Reply:
Ah, thanks. I was thinking that it's only the kind of object being used and so the second call was not needed, but it really should be updated. I moved the update into the else block so that it is only done when needed. For what it's worth, I added an assert to see if the value order ever did change and it never got triggered by the tests.